graphql_compiler/schema_transformation/restrict_schema.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from copy import copy
+
+from graphql import build_ast_schema
+from graphql.language.visitor import REMOVE, Visitor, visit
+
+from ..ast_manipulation import get_ast_with_non_null_and_list_stripped
+from .utils import SchemaStructureError, get_query_type_name, get_scalar_names
+
+
+def restrict_schema(schema_ast, types_to_keep):
```

Comment:
In the foreseeable future, we might have a similar function that restricts _which fields_ are visible on certain types, so it might be good to make sure the naming is ready for that. Maybe `restrict_schema_types` instead, so that `restrict_schema_fields` can be added later?